fastly/client.go

```diff
@@ -742,20 +738,22 @@ func stringToTimeHookFunc() mapstructure.DecodeHookFunc {
 		t reflect.Type,
 		data any,
 	) (any, error) {
-		if f.Kind() != reflect.String {
+		// Ensure the source is a string and the target is time.Time
+		if f.Kind() != reflect.String || t != reflect.TypeOf(time.Time{}) {
 			return data, nil
 		}
-		if t != reflect.TypeOf(time.Now()) {
-			return data, nil
+
+		// Attempt parsing in RFC3339 format
+		str, _ := data.(string) // Safe type assertion; guaranteed by f.Kind()
+		if v, err := time.Parse(time.RFC3339, str); err == nil {
+			return v, nil
 		}
 
-		// Convert it by parsing
-		v, err := time.Parse(time.RFC3339, data.(string))
-		if err != nil {
-			// DictionaryInfo#get uses it's own special time format for now.
```

Comment:
This comment should be preserved, as it explains why the second parse attempt is necessary.